### ISSUE TITLE
Naming Conventions: Make Task List Files More Modular (sed and phonics puns)

### ISSUE THREAD

**Issue by doctortheisen** [cumulativerecords/trainaba-v1-ed2#33]
1. How to do this, mechanically?
Use sed to redirect text from specific lines:

`sed -n '1,5p' origin-file > destination-file`

2. What are the naming conventions?

The hard decision is whether to break up the definitions and description section. I faced this decision in 2017 for the first versions of the Moodle course packs and decided to roll it into one section called "concept." The only other ideas here involve dividing up a new definitions file but that can come later if needed. The definitions already pull from a definitions.sty file so it's easy to keep them organized anyway. I'll call info.tex because it's info about at the module. It's also short enough to type without a cryptic phonic pun. My puns for "concept" look too weird.

The phonic puns are stupid but I like them. The phonics puns honor my mom's old book, "C-D-B".

SECTION                       NEW FILE NAME
definitions            =         nfo.tex
description           =         (same)
examples              =         xmplz.tex
assessment          =         ssmt.tex     
relevant literature =        lit.tex
related lessons     =        lssnz.tex.    /* phonics pun again */

**Comment by doctortheisen** [comment]
For readability, I should make these names readable. 
But it's too fun. Besides, if someone gets this deep into the project they should be able to find their way around.

**Comment by doctortheisen** [comment]
Remember sed conventions:

A single angle bracket creates a file. The double angle bracket appends that file.

**Comment by doctortheisen** [comment]
Workflow for implementation: 

1. Open a terminal to current task list module (ex: fiveaOne):
`cd ~/tex/chapters/fifth-ed/tl-modules/fiveaOne`

2. Open the Finder to see file structure. This is how to visually inspect that files are created as expected and preview contents.

3. Open the body.tex file in the current task list module.

4. Add the opening lines with a dummy section name entry (ex: fiveaTwo):
```
\clearpage \section{\fiveaTwo{}}
\input{tex/chapters/fifth-ed/tl-modules/fiveaTwo/nfo}
\input{tex/chapters/fifth-ed/tl-modules/fiveaTwo/xmplz}
\input{tex/chapters/fifth-ed/tl-modules/fiveaTwo/ssmt}
\input{tex/chapters/fifth-ed/tl-modules/fiveaTwo/lit}
\input{tex/chapters/fifth-ed/tl-modules/fiveaTwo/lssnz}

%\clearpage \section[\fourFKSeven{}]{\fourFKSeven{}%
%             \sectionmark{FK-07 Environmental (as opposed...}}
%\sectionmark{FK-07 Environmental (as opposed...}
```

5. Update sed scripts using ranges. Repeat for each task list in the body.tex module. The line numbers start with 11 because I delete the content after streaming it to the new modular file location.

Examples:

```
sed -n '11,20p' body.tex >> nfo.tex
sed -n '11,18p' body.tex >> xmplz.tex
sed -n '11,18p' body.tex >> ssmt.tex
sed -n '11,19p' body.tex >> lit.tex
sed -n '11,18p' body.tex >> lssnz.tex
```

6. Later, I may want to add, rearrange, or repurpose parts of a task list module. It is very easy to make such changes with this modular structure.

**Comment by doctortheisen** [comment]
Created a program to handle this issue. 

Name: ./make-subfiles
Subroutine: ./ranger

The program ./make-subfiles takes an argument (task list folder to make into subfiles).
It pops up the body.tex file in a text editor window, showing line numbers and content so I can assess the ranges visually. These ranges are random. 
It uses a menu system, asking me questions at every step.
I enter the subfile to create, as well as the range of lines for that subfile.
The program creates the subfile.
It loops through the menu an arbitrary number of times, to accommodate subfiles made from multiple fourth edition task list entries.
At the end, it gives the option of overwriting the body.tex file with an index or to skip (in case I made a mistake and want to review the file). 


An ideal alternative to this program would be to create a sed program that copies text between tags, sending it to subfiles. It could just loop through a task list. The whole thing would be done in one program run. However, I did not use tags when writing the 4th ed task list, so this is not helpful. It's a bigger pain to visually open every file, punch in tags, and write a sed program to process tags. Might as well just run this program.
